cardano-api/src/Cardano/Api/ProtocolParameters.hs

```diff
@@ -281,13 +285,16 @@ data ProtocolParameters =
     }
   deriving (Eq, Generic, Show)
 
+parseRationalJSON :: Aeson.Value -> Parser Rational
+parseRationalJSON v = parseJSON v <|> (toRational @Scientific <$> parseJSON v)
```

Comment:
I think we might not need this special case on the parsing side. The existing `FromJSON` instance for `Ratio` (and hence `Rational`) already handles number style or object style:
```
instance (FromJSON a, Integral a) => FromJSON (Ratio a) where
    parseJSON (Number x)
      | exp10 <= 1024
      , exp10 >= -1024 = return $! realToFrac x
      | otherwise      = prependContext "Ratio" $ fail msg
      where
        exp10 = base10Exponent x
        msg = "found a number with exponent " ++ show exp10
           ++ ", but it must not be greater than 1024 or less than -1024"
    parseJSON o = objParser o
      where
        objParser = withObject "Rational" $ \obj -> do
            numerator <- obj .: "numerator"
            denominator <- obj .: "denominator"
            if denominator == 0
            then fail "Ratio denominator was 0"
            else pure $ numerator % denominator
    {-# INLINE parseJSON #-}
```
https://hackage.haskell.org/package/aeson-1.5.6.0/src/src/Data/Aeson/Types/FromJSON.hs

So this looks like it ought to "just work". Perhaps you can check if it does indeed work.

Reply:
```
λ> decode "1.0" :: Maybe Rational
Just (1 % 1)
```

Reply:
Fixed!